st3/__init__.py

```diff
@@ -0,0 +1,125 @@
+from ..common import create_launch_command
+from ..common import get_java_path
+from ..common import NewScalaFileCommand
+from ..common import prepare_server_properties
+from LSP.plugin.core.collections import DottedDict
+from LSP.plugin.core.handlers import LanguageHandler
+from LSP.plugin.core.protocol import Location, Response
+from LSP.plugin.core.settings import ClientConfig, LanguageConfig
+import os
+import sublime
+
+server_name = 'LSP-metals'
+settings_file = 'LSP-metals.sublime-settings'
+
+
+class LspMetalsPlugin(LanguageHandler):
+    def __init__(self):
+        missing_java_home = "Unable to find a Java 8 or Java 11 installation on this computer. Please set `java_home` in the settings."
+        plugin_settings = sublime.load_settings(settings_file)
+        server_version = plugin_settings.get('server_version')
+        server_properties = prepare_server_properties(plugin_settings.get('server_properties', []))
+        java_path = get_java_path(plugin_settings)
+
+        launch_command = []
+        if java_path and server_version :
+            launch_command = create_launch_command(java_path, server_version, server_properties)
+        if java_path is None:
+            sublime.error_message(missing_java_home)
+
+        try:
+            # LSP 1.0
+            language = LanguageConfig("scala")
+        except TypeError:
+            # Ancient LSP
+            language = LanguageConfig(
+                language_id="scala",
+                scopes=["source.scala"],
+                syntaxes=["Packages/Scala/Scala.sublime-syntax"])
+
+        metals_config = ClientConfig(
+            name=server_name,
+            binary_args=launch_command,
+            tcp_port=None,
+            languages=[language],
+            enabled=True,
+            init_options=DottedDict(plugin_settings.get("initializationOptions")),
```

Comment:
I added DottedDict. I'm not sure how this previously even worked without DottedDict.